src/tasks/worker.ts

```diff
@@ -0,0 +1,263 @@
+	'is scheduled with-cron expression': 'is_scheduled_with__cron_expression',
+	'attempt count': 'attempt_count',
+	'attempt limit': 'attempt_limit',
+	priority: 'priority',
+	'is created by-actor': 'is_created_by__actor',
+})
+	.map(([key, value]) => `t."${key}" AS "${value}"`)
+	.join(', ');
+
+// The worker is responsible for executing tasks in the queue. It listens for
+// notifications and polls the database for tasks to execute. It will execute
+// tasks in parallel up to a certain concurrency limit.
+export class Worker {
+	public handlers: Record<string, TaskHandler> = {};
+	private readonly concurrency: number;
+	private readonly interval: number;
+	private client: PinejsClient;
+	private executing = 0;
+
+	constructor(client: PinejsClient) {
+		this.client = client;
+		this.concurrency = tasksEnv.queueConcurrency;
+		this.interval = tasksEnv.queueIntervalMS;
+	}
+
+	// Check if instance can execute more tasks
+	private canExecute(): boolean {
+		return (
+			this.executing < this.concurrency && Object.keys(this.handlers).length > 0
+		);
+	}
+
+	// Execute a task
+	private async execute(tx: Db.Tx, task: PartialTask): Promise<void> {
+		this.executing++;
+		try {
+			// Get specified handler
+			const handler = this.handlers[task.is_executed_by__handler];
+			const startedOnTime = new Date();
+			if (handler == null) {
+				await this.finalize(
+					tx,
+					task,
+					startedOnTime,
+					'failed',
+					'Matching task handler not found',
+				);
+				return;
+			}
+
+			// Validate parameters before execution so we can fail early if
+			// the parameter set is invalid. This can happen if the handler
+			// definition changes after a task is added to the queue.
+			if (
+				handler.validate != null &&
+				!handler.validate(task.is_executed_with__parameter_set)
+			) {
+				await this.finalize(
+					tx,
+					task,
+					startedOnTime,
+					'failed',
+					`Invalid parameter set: ${ajv.errorsText(handler.validate.errors)}`,
+				);
+				return;
+			}
+
+			// Execute handler
+			let status: TaskStatus = 'queued';
+			let error: string | undefined;
+			try {
+				await sbvrUtils.db.transaction(async (handlerTx) => {
+					const results = await handler.fn({
+						api: new PinejsClient({
+							passthrough: {
+								tx: handlerTx,
+							},
+						}),
+						params: task.is_executed_with__parameter_set ?? {},
+						tx: handlerTx,
+					});
+					status = results.status;
+					error = results.error;
+					if (results.status !== 'succeeded' && !handlerTx.isClosed()) {
+						await handlerTx.rollback();
+					}
+				});
+			} catch (err) {
+				// Ignore closed/rollback errors
+				if (!(err instanceof TransactionClosedError)) {
+					throw err;
+				}
+			} finally {
+				// Update task with results
+				await this.finalize(tx, task, startedOnTime, status, error);
+			}
+		} catch (err) {
+			// This shouldn't happen, but if it does we want to log and kill the process
+			console.error('Task execution failed:', err);
```

Comment:
Can you also add some details about the task, eg the task handler name, since I expect it'll help significantly with debugging